include/seqan3/std/ranges

```diff
@@ -280,4 +295,646 @@ constexpr bool enable_view<type> = static_cast<bool>(seqan3::detail::enable_view
+ * |---------------------------------|:--------------------------------------------------:|
+ * | std::ranges::InputRange         | *preserved*                                        |
+ * | std::ranges::ForwardRange       | *preserved*                                        |
+ * | std::ranges::BidirectionalRange | *preserved*                                        |
+ * | std::ranges::RandomAccessRange  | *preserved*                                        |
+ * | std::ranges::ContiguousRange    | *preserved*                                        |
+ * |                                 |                                                    |
+ * | std::ranges::ViewableRange      | *guaranteed*                                       |
+ * | std::ranges::View               | *guaranteed*                                       |
+ * | std::ranges::SizedRange         | *preserved*                                        |
+ * | std::ranges::CommonRange        | *preserved*                                        |
+ * | std::ranges::OutputRange        | *preserved*                                        |
+ * | seqan3::const_iterable_concept  | *preserved*                                        |
+ * |                                 |                                                    |
+ * | seqan3::reference_t             | seqan3::value_type_t<it_t>                         |
+ *
+ * Preservation in this table refers to the properties of the iterator/sentinel pair.
+ *
+ * See the \link view view submodule documentation \endlink for detailed descriptions of the view properties.
+ *
+ * ### STD module
+ *
+ * This entity will likely be part of C++20 or C++23. It's API will track current proposals and not be stable
+ * within SeqAn3 releases. It is implemented via the range-v3 library or the standard library (if available).
+ * Should it become clear that it will not become part of a future standard, it will migrate to a regular SeqAn3
+ * module.
+ *
+ * ### Example
+ *
+ * \snippet test/snippet/std/view/subrange.cpp example
+ * \hideinitializer
+ */
+template <std::Iterator it_t, std::Sentinel<it_t> sen_t>
+using subrange = ::ranges::subrange<it_t, sen_t>;
+
+} // namespace std::ranges
+
+// -----------------------------------------------------------------------------------------------------
+// View adaptor
+// -----------------------------------------------------------------------------------------------------
+
+namespace std::ranges::view
+{
+
+/*!\brief A range adaptor that forwards views and "view-wraps" containers (you will likely not need to use this unless defining a new view).
+ * \tparam    urng_t The type of the range being processed. See below for requirements.
+ * \param[in] urange The range being processed.
+ * \returns A view over the elements of the underlying range.
+ * \ingroup ranges
+ *
+ * ### View properties
+ *
+ * This view is **source-only**, it can only be at the beginning of a pipe of range transformations. The
```

Comment:
Really? Can't I have it on the right hand side? (even if it does not make much sense)

Reply:
> Really?

nope